WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderReadingPreferencesViewModel.kt

```diff
@@ -65,15 +71,24 @@ class ReaderReadingPreferencesViewModel @Inject constructor(
         readingPreferencesTracker.trackItemTapped(fontSize)
     }
 
-    fun saveReadingPreferencesAndClose() {
+    /**
+     * An exit action has been triggered by the user. This means that we need to save the current preferences and emit
+     * the close event, so the dialog is dismissed.
+     */
+    fun onExitActionClick() {
         launch {
-            val currentPreferences = currentReadingPreferences.value
-            val isDirty = currentPreferences != originalReadingPreferences
-            if (isDirty) {
-                saveReadingPreferences(currentPreferences)
-                readingPreferencesTracker.trackSaved(currentPreferences)
-            }
-            _actionEvents.emit(ActionEvent.Close(isDirty))
+            saveReadingPreferencesInternal()
+            _actionEvents.emit(ActionEvent.Close)
+        }
+    }
+
+    /**
+     * The bottom sheet has been hidden by the user, which means the dismiss process is already on its way. All we need
+     * to do is save the current preferences.
+     */
+    fun onBottomSheetHidden() {
```

Comment:
I'm not a big fan of that approach since doing that causes the function to return a `Job`, which is not really what I'm going for and it would be just a side-effect of trying to make the code look prettier.

I get that using `= launch` is kinda common practice. Still, I personally think that the function signature should better match the intention of the code, so if no callers need the returned `Job` (and they shouldn't as this function is public and the coroutine is an implementation detail relevant to the `ViewModel` only) I prefer "returning" `Unit`.